### PR TITLE
470 history db record attribute access

### LINKED ISSUE #470: [Bug]: historical database records can not be accessed via keywords
### What version of Datahike are you using?

0.4.1480

### What version of Java are you using?

openjdk version "11.0.13" 2021-10-19

### What operating system are you using?

Ubuntu 20.04.3 LTS

### What database EDN configuration are you using?

any with `keep-history?` set to `true`

### Describe the bug

When interacting with any historical database record `HistoricalDB`, `AsOfDB`, `SinceDB`, one cannot access any attributes using keywords.

### What is the expected behaviour?

Access the db record attributes without any exceptions.

### How can the behaviour be reproduced?

```clojure
(require '[datahike.api :as d])

(def cfg {:store              {:backend :mem
                               :id      "DEV"}
          :keep-history?      true
          :schema-flexibility :read})

(do
  (d/delete-database cfg)
  (d/create-database cfg))

(def conn (d/connect cfg))

(d/transact conn [{:name "Alice"}
                  {:name "Bob"}])

(d/transact conn [{:name "Charlie"}])

(let [db (d/as-of @conn (inc datahike.constants/tx0))]
  (:origin-db db))
;; => Execution error (UnsupportedOperationException) at datahike.db.AsOfDB/getLookupThunk (db.cljc:563).
;;    getLookupThunk is not supported on AsOfDB
```

### PR DESCRIPTION
**Pull request by kordano**
#### SUMMARY
Fixes #470 

##### Bugfix
- [x] Related issues linked using `fixes #number`
- [x] Integration tests added
- [x] Formatting checked
- [x] `CHANGELOG.md` updated

### CODE REVIEW

**Review by TimoKramer — DISMISSED** [review]
I don't get why these interfaces with the UnsupportedOperationException were there in the first place?! :thinking:

File: test/datahike/test/time_variance_test.cljc [hunk 1]
```diff
@@ -323,3 +323,26 @@
     (d/release conn)
     (d/delete-database cfg)))
 ;; => #'datahike.test.time-variance/test-no-duplicates-with-cardinality-many
+
+;; https://github.com/replikativ/datahike/issues/470
+(deftest test-history-record-attribute-access
+  (let [cfg {:store {:backend :mem
+                     :id "test-history-record-attribute-access"}
+             :keep-history? true
+             :schema-flexiibility :read
```
Comment (jsmassa) [inline test/datahike/test/time_variance_test.cljc:332]: typo:  ` :schema-flexiibility` -> ` :schema-flexibility`

**Review by jsmassa — COMMENTED** [review]
I don't see a reason either why they where in there, especially because it works as shown here... weird...
Except for the typo, I can't see anything wrong with this.

File: test/datahike/test/time_variance_test.cljc [hunk 2]
```diff
@@ -323,3 +323,26 @@
     (d/release conn)
     (d/delete-database cfg)))
 ;; => #'datahike.test.time-variance/test-no-duplicates-with-cardinality-many
+
+;; https://github.com/replikativ/datahike/issues/470
+(deftest test-history-record-attribute-access
+  (let [cfg {:store {:backend :mem
+                     :id "test-history-record-attribute-access"}
```
[the remark this answers is not in the document]
Comment (kordano) [reply test/datahike/test/time_variance_test.cljc:330]: Didn't know that. Probably would make sense to have that as default so one might set his one ID.

**Review by jsmassa — APPROVED** [review]
Thanks for the fix

**Pull request merged**